sympy/core/relational.py

```diff
@@ -627,41 +627,7 @@ def _eval_relation(cls, lhs, rhs):
         return _sympify(lhs == rhs)
 
     def _eval_rewrite_as_Add(self, L, R, evaluate=True, **kwargs):
-        """
-        return Eq(L, R) as L - R. To control the evaluation of
-        the result set pass `evaluate=True` to give L - R;
-        if `evaluate=None` then terms in L and R will not cancel
-        but they will be listed in canonical order; otherwise
-        non-canonical args will be returned. If one side is 0, the
-        non-zero side will be returned.
-
-        Examples
-        ========
-
-        >>> from sympy import Eq, Add
-        >>> from sympy.abc import b, x
-        >>> eq = Eq(x + b, x - b)
-        >>> eq.rewrite(Add)
-        2*b
-        >>> eq.rewrite(Add, evaluate=None).args
-        (b, b, x, -x)
-        >>> eq.rewrite(Add, evaluate=False).args
-        (b, x, b, -x)
-        """
-        from .add import _unevaluated_Add, Add
-        if L == 0:
-            return R
-        if R == 0:
-            return L
-        if evaluate:
-            # allow cancellation of args
-            return L - R
-        args = Add.make_args(L) + Add.make_args(-R)
-        if evaluate is None:
-            # no cancellation, but canonical
-            return _unevaluated_Add(*args)
-        # no cancellation, not canonical
-        return Add._from_args(args)
+        assert None  # XXX are we going to deprecate?
```

Comment:
Nothing else has `_eval_rewrite_as_Add` as far as I can tell so it should be okay to deprecate. It is not hard to change code to be compatible with this either because it is just `eq.lhs - eq.rhs` which has always worked in previous versions.

I would say let's make sure that SymPy itself does not use this but then leave it in with a DeprecationWarning. If ever we need to implement `_eval_rewrite_as_Add` for something else we should remove this completely though.